cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DocumentStore.java

```diff
@@ -65,10 +65,30 @@
     private static final Logger logger = Logger.getLogger(DocumentStore.class.getCanonicalName());
 
     private DocumentStore(File location, KeyProvider keyProvider) throws DatastoreException, IOException, SQLException {
-        this.location = location;
-        this.databaseName = location.toString();
-        this.database = new DatabaseImpl(location, keyProvider);
-        this.query = new IndexManagerImpl(database);
+        try {
+            this.location = location;
+            this.databaseName = location.toString();
+            this.database = new DatabaseImpl(location, keyProvider);
+            this.query = new IndexManagerImpl(database);
+        } catch (DatastoreException e) {
```

Comment:
As discussed, can't be done without changing the constructor signature to `throws Exception`